sql/core/src/test/scala/org/apache/spark/sql/execution/command/DDLSuite.scala

```diff
@@ -1816,4 +1816,127 @@ class DDLSuite extends QueryTest with SharedSQLContext with BeforeAndAfterEach {
       }
     }
   }
+
+  test("insert data to a data source table which has a not existed location should succeed") {
+    withTable("t") {
+      withTempDir { dir =>
+        spark.sql(
+          s"""
+              |CREATE TABLE t(a string, b int)
+              |USING parquet
+              |OPTIONS(path "file:${dir.getCanonicalPath}")
```

Comment:
- First, you do not need to add the `file:`
- Second, you still need to adjust the indent.

```Scala
        spark.sql(
          s"""
             |CREATE TABLE t(a int, b int, c int, d int)
             |USING parquet
             |PARTITIONED BY(a, b)
             |LOCATION '$dir'
           """.stripMargin)
        val expectedPath = dir.getAbsolutePath.stripSuffix("/")
```